sdk/formrecognizer/ai-form-recognizer/src/formRecognizerClient.ts

```diff
@@ -387,21 +378,43 @@ export class FormRecognizerClient {
     if (!modelId) {
       throw new RangeError("Invalid model id");
     }
-    const analyzePollerClient: RecognizeCustomFormPollerClient = {
-      beginRecognize: (
-        body: FormRecognizerRequestBody | string,
-        modelId: string,
-        contentType?: FormContentType,
-        analyzeOptions: RecognizeFormsOptions = {}
-      ) => recognizeCustomFormInternal(this.client, body, contentType, analyzeOptions, modelId!),
-      getRecognizeResult: (resultId: string, options: { abortSignal?: AbortSignalLike }) =>
-        this.getRecognizedForm(modelId, resultId, options)
-    };
 
-    const poller = new BeginRecognizeCustomFormPoller({
-      client: analyzePollerClient,
+    const { span } = makeSpanner("FormRecognizerClient-beginRecognizeCustomForms", {
+      ...options,
+      includeTextDetails: options.includeFieldElements
+    });
+
+    const poller = new FormRecognitionPoller({
       modelId,
-      source: form,
+      createOperation: span("customFormsInternal", async (finalOptions) => {
+        const requestBody = await toRequestBody(form);
+        const contentType = finalOptions.contentType ?? (await getContentType(requestBody));
+        const { operationLocation } = await this.client.analyzeWithCustomModel(
+          modelId,
+          contentType!,
+          requestBody as Blob | ArrayBuffer | ArrayBufferView,
+          operationOptionsToRequestOptionsBase(finalOptions)
+        );
+
+        if (!operationLocation) {
+          throw new Error(
+            "The service did not respond with an 'operationLocation' to retrieve results."
+          );
+        } else {
+          const lastSlashIndex = operationLocation.lastIndexOf("/");
+          return operationLocation.substring(lastSlashIndex + 1);
+        }
+      }),
```

Comment:
I wonder whether this piece of common code is worth to extract out

Reply:
I abstracted this out into a helper function. When I add invoice support I think I can maybe add some extra logic to reduce the duplication of body parsing as well.